runtime/compiler/env/J9SharedCache.cpp

```diff
@@ -885,7 +885,15 @@ TR_J9SharedCache::rememberClass(J9Class *clazz, const AOTCacheClassChainRecord *
    chainData = findChainForClass(clazz, key, keyLength);
    if (chainData != NULL)
       {
-      LOG(1, "\tchain exists (%p) so nothing to store\n", chainData);
+      if (classMatchesCachedVersion(clazz, chainData))
+         {
+         LOG(1, "\tchain exists (%p) so nothing to store\n", chainData);
```

Comment:
These messages feel a bit cryptic to me.

In the matching case, I suggest:
   class and class chain found are identical so rememberClass returns the class chain

In the failing case, I suggest:
   chain found does not match current class, cannot use class chain so rememberClass returns NULL